linode/linode-cli
# new: Support specifying complex list arguments as JSON

## 📝 Description

This change adds support for specifying complex lists (e.g. `interfaces`) as a single JSON string rather than a series of arguments. This should improve the UX when writing scripts that may use these complex structures.

This pull request also makes various changes to generalize parent/child logic rather than explicitly relating it to lists.

This PR originally included improvements to the command help pages but I decided to break it out into a separate PR because of how big this PR was getting 🙂 

## ✔️ How to Test

The following test steps assume you have pulled down this PR and run `make install`.

### Integration Testing

```bash
make testint
```

### Unit Testing

```bash
make testunit
```

### Manual Testing 

#### Complex Lists as JSON


1. Attempt to create a Linode with the `interfaces` argument defined as JSON:

```bash
linode-cli linodes create \
  --label test-instance \
  --region us-mia \
  --image linode/alpine3.19 \
  --root_pass 'testp4ssw0rd!!!!!!23123' \
  --type g6-nanode-1 \
  --interfaces '[{"purpose": "public"}, {"purpose": "vlan", "label": "test-vlan"}]' \
  --debug
```

2. Ensure the interfaces JSON is properly represented in the debug request body output.
3. Navigate to the instance in Cloud Manager and ensure it has been configured as expected.
